File: src/app/models/DigitalTwin.js


// Define collection and schema for DigitalTwin
export interface DigitalTwin {
    twinId:
	type : string
    desiredStateVersion:
	type : number
    reportedStateVersion:
	type : number
    lastSyncAt:
	type : Date
    Device:
	type : Schema.Types.ObjectId
    Gateway:
	type : Schema.Types.ObjectId
    Template:
	type : Schema.Types.ObjectId
    ChangeEvents:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TwinChangeEvent' }]
#
    collection: 'digitalTwins'
}

File: iotonaspdotnet/Domain/DigitalTwin.cs
using iotonaspdotnet.Contracts;

namespace iotonaspdotnet.Domain;

public class DigitalTwin
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? DigitaltwinId { get; set; }
 public virtual string? TwinId { get; set; }
 public virtual int? DesiredStateVersion { get; set; }
 public virtual int? ReportedStateVersion { get; set; }
 public virtual DateTime? LastSyncAt { get; set; }
public virtual IoTDevice? Device { get; set; }
public virtual Gateway? Gateway { get; set; }
public virtual TwinTemplate? Template { get; set; }
public virtual TwinChangeEvent? ChangeEvents { get; set; }

    public static DigitalTwin FromRequest(DigitalTwinRequest request) {
        return new DigitalTwin {
            Id = request.Id,
            TwinId = request.TwinId,
            DesiredStateVersion = request.DesiredStateVersion,
            ReportedStateVersion = request.ReportedStateVersion,
            LastSyncAt = request.LastSyncAt,
        };
    }
}
